### PR TITLE
Feature/cc 1466

### PR DESCRIPTION
**Pull request by jplouis**

### CODE REVIEW

**Comment by jplouis** [comment]
retest this

File: commons/pool/pool_test.go [hunk 1]
```diff
@@ -0,0 +1,232 @@
+// Copyright 2014 The Serviced Authors.
```
Comment (tgjonestx) [inline commons/pool/pool_test.go]: should be 2015

File: commons/queue/queue_test.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
+// Copyright 2014 The Serviced Authors.
```
Comment (tgjonestx) [inline commons/queue/queue_test.go]: Should be 2015

File: commons/queue/queue_test.go [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+func (s *MySuite) TestNewQueue(c *C) {
+	var q Queue
+	_, err := NewChannelQueue(-1)
+	c.Assert(err, NotNil)
+
+	_, err = NewChannelQueue(0)
+	c.Assert(err, NotNil)
+
+	q, err = NewChannelQueue(1)
+	c.Assert(q.Capacity(), Equals, int32(1))
```
Comment (tgjonestx) [inline commons/queue/queue_test.go]: And check q.Size() == 0

File: commons/queue/queue_test.go [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+}
+
+func (s *MySuite) TestOffer(c *C) {
+	cap := 32
+	q, _ := NewChannelQueue(cap)
+	if q == nil {
+		c.Fail()
+	}
+	for i := 0; i < cap*2; i++ {
+		q.Offer(i)
```
Comment (tgjonestx) [inline commons/queue/queue_test.go]: Check return value of Offer() to verify true/false relative to capacity

File: commons/pool/pool_test.go [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+	start := time.Now()
+	x, err := p.BorrowWait(waitTime)
+	c.Assert(x, IsNil)
+	c.Assert(err, Equals, ErrItemUnavailable)
+	elapsed := time.Now().Sub(start)
+	if elapsed < waitTime {
+		c.Errorf("elapsed wait less than %s", waitTime)
+	}
+
+	wg := sync.WaitGroup{}
```
Comment (tgjonestx) [inline commons/pool/pool_test.go]: pls add a comment here to explain the need for the waitgroup

**Comment by tgjonestx** [comment]
:+1:

**Pull request merged**